### PR TITLE
[PW-7587] Remove IsApiKeyRequired

### LINKED ISSUE #679: [PW-7587] Checkout API, Basic Authentication (username + password) does not work using synchronous methods
**Describe the bug**
Using synchronous endpoints in the Checkout API (and possible others, have not checked) will result in 401 if the client is created using basic auth.

I checked the code and the reason they work differently is that ServiceResource.cs handles the IsApiKeyRequired flag differently.
The asyncversion always sends false to the GetHttpWebRequest method.

https://github.com/Adyen/adyen-dotnet-api-library/blob/ab8f628d461295f44ae96f4486ea617a1a9eb1ac/Adyen/Service/ServiceResource.cs#L43-L55

The GetHttpWebRequest has a possible bug (not sure what the flag is supposed to do exactly so this might be WAD) that never uses basic auth if isApiKeyRequired is true.
https://github.com/Adyen/adyen-dotnet-api-library/blob/ab8f628d461295f44ae96f4486ea617a1a9eb1ac/Adyen/HttpClient/HttpURLConnectionClient.cs#L187-L192

**To Reproduce**
Initialize the client and request any endpoint using the non-async version of the call

**Expected behavior**
It should work just the same as the async version

**Workaround**
If anyone else runs into this issue before it's fixed, a workaround is to explicitly set IsApiKeyRequired to false when initializing the Checkout Service.
```cs
var checkout = new Checkout(client)
{
    IsApiKeyRequired = false
};
```

### PR DESCRIPTION
**Pull request by jillingk**
**Description**
Fixes #679

### CODE REVIEW

**Review by AlexandrosMor — DISMISSED** [review]

File: Adyen.IntegrationTest/CheckoutTest.cs [hunk 1]
```diff
@@ -124,12 +127,38 @@ public void PaymentLinksSuccessTest()
...
+            Assert.IsNotNull(createPaymentLinkResponse);
+            Assert.IsNotNull(createPaymentLinkResponse.Url);
+            Assert.IsNotNull(createPaymentLinkResponse.Amount);
+            Assert.IsNotNull(createPaymentLinkResponse.Reference); 
+            Assert.IsNotNull(createPaymentLinkResponse.ExpiresAt);
+            }
+        
+        private void PaymentLinksGetSuccessTest(string Id)
+        {   
+
```
Comment (wboereboom) [inline Adyen.IntegrationTest/CheckoutTest.cs:141]:
```suggestion
```

File: Adyen.IntegrationTest/CheckoutTest.cs [hunk 2]
```diff
@@ -124,12 +127,38 @@ public void PaymentLinksSuccessTest()
...
             };
             var createPaymentLinkResponse = _checkout.PaymentLinks(createPaymentLinkRequest);
+            PaymentLinksGetSuccessTest(createPaymentLinkResponse.Id);
+            PaymentLinksPatchSuccessTest(createPaymentLinkResponse.Id);
+            Assert.IsNotNull(createPaymentLinkResponse);
+            Assert.IsNotNull(createPaymentLinkResponse.Url);
+            Assert.IsNotNull(createPaymentLinkResponse.Amount);
+            Assert.IsNotNull(createPaymentLinkResponse.Reference); 
+            Assert.IsNotNull(createPaymentLinkResponse.ExpiresAt);
+            }
```
Comment (wboereboom) [inline Adyen.IntegrationTest/CheckoutTest.cs:137]:
```suggestion
        }
```

File: Adyen.IntegrationTest/CheckoutTest.cs [hunk 3]
```diff
@@ -124,12 +127,38 @@ public void PaymentLinksSuccessTest()
...
+            var updatePaymentLinksRequest = new UpdatePaymentLinkRequest(status: UpdatePaymentLinkRequest.StatusEnum.Expired);
+            
+            var createPaymentLinkResponse = _checkout.PatchPaymentLinks(updatePaymentLinksRequest, Id);
             Assert.IsNotNull(createPaymentLinkResponse);
             Assert.IsNotNull(createPaymentLinkResponse.Url);
             Assert.IsNotNull(createPaymentLinkResponse.Amount);
             Assert.IsNotNull(createPaymentLinkResponse.Reference);
             Assert.IsNotNull(createPaymentLinkResponse.ExpiresAt);
         }
+
```
Comment (wboereboom) [inline Adyen.IntegrationTest/CheckoutTest.cs:161]:
```suggestion
```

File: Adyen.IntegrationTest/PaymentTest.cs [hunk 4]
```diff
@@ -69,12 +70,12 @@ public void ApiIdemptotencyKeyFailTest()
...
                 var paymentResult2 = CreatePaymentResultWithIdempotency("AUTH_IDEMPOTENCY_KEY_AUTHOR");
             }
-            catch (HttpClient.HttpClientException ex)
+            catch (HttpClientException ex)
             {
                 Assert.AreEqual(ex.Code, 403);
             }
         }
-
+
```
Comment (wboereboom) [inline Adyen.IntegrationTest/PaymentTest.cs:78]:
```suggestion
```

File: Adyen.Test/BaseTest.cs [hunk 5]
```diff
@@ -349,11 +368,15 @@ protected Client CreateMockTestClientPosLocalApiRequest(string fileName)
             //Create a mock interface
             var clientInterfaceMock = new Mock<IClient>();
             clientInterfaceMock.Setup(x => x.Request(It.IsAny<string>(),
-                It.IsAny<string>(), config, It.IsAny<bool>(),
-                It.IsAny<RequestOptions>()
+                It.IsAny<string>(),
+                It.IsAny<RequestOptions>(), null
```
Comment (wboereboom) [inline Adyen.Test/BaseTest.cs:372]:
```suggestion
                It.IsAny<RequestOptions>(),
                null
```

File: Adyen.Test/PayoutTest.cs [hunk 6]
```diff
@@ -105,19 +107,20 @@ public void PayoutSuccessTest()
             Assert.AreEqual("8814689190961342", result.PspReference);
             Assert.AreEqual("12345", result.AuthCode);
         }
-
+
```
Comment (wboereboom) [inline Adyen.Test/PayoutTest.cs:110]:
```suggestion
```

File: Adyen.Test/PayoutTest.cs [hunk 7]
```diff
@@ -105,19 +107,20 @@ public void PayoutSuccessTest()
...
+            var payout = new Payout(client);
+            var request = new PayoutRequest();
             try
             {
-                payout.PayoutSubmit(request);
+                Console.WriteLine(payout.PayoutSubmit(request));
             }
             catch (HttpClient.HttpClientException e)
             {
+                Console.WriteLine(e.ResponseBody);
```
Comment (wboereboom) [inline Adyen.Test/PayoutTest.cs:123]: Still some console statements in this PR

File: Adyen/HttpClient/HttpClientExtension.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
+#region License
+// /*
```
Comment (wboereboom) [inline Adyen/HttpClient/HttpClientExtension.cs:2]: We do not want to use these ASCII headers anymore

File: Adyen/Service/Checkout.cs [hunk 9]
```diff
@@ -198,7 +198,7 @@ public PaymentLinkResponse PaymentLinks(CreatePaymentLinkRequest createPaymentLi
             var jsonResponse = _paymentLinksResult.Request(jsonRequest);
             return JsonConvert.DeserializeObject<PaymentLinkResponse>(jsonResponse);
         }
-
+
```
Comment (wboereboom) [inline Adyen/Service/Checkout.cs:201]:
```suggestion
```

File: Adyen/Service/Checkout.cs [hunk 10]
```diff
@@ -210,6 +210,64 @@ public async Task<PaymentLinkResponse> PaymentLinksAsync(CreatePaymentLinkReques
...
+            var jsonResponse = await paymentLinks.RequestAsync(null, null, HttpMethod.Get);
+            return JsonConvert.DeserializeObject<PaymentLinkResponse>(jsonResponse);
+        }
+        /// <summary>
+        /// PATCH /paymentsLinks API call 
+        /// </summary>
+        /// <param name="updatePaymentLinkRequest"></param>
+        /// <param name="linkId"></param>
+        /// <returns>PaymentLinkResponse</returns>
+
```
Comment (wboereboom) [inline Adyen/Service/Checkout.cs:245]: whiteline shut be put above method description

**Review by wboereboom — CHANGES_REQUESTED** [review]

File: Adyen/HttpClient/HttpClientExtension.cs (hunk 8 above)
Comment (jillingk) [reply Adyen/HttpClient/HttpClientExtension.cs:2]: Good point, I started removing them in these PR's but @AlexandrosMor suggested I remove all of them in a separate PR.

**Review by michaelpaul — APPROVED** [review]

**Review by wboereboom — APPROVED** [review]

**Pull request merged**